ping/ping_common.c

```diff
@@ -401,6 +399,10 @@ int pinger(struct ping_rts *rts, ping_func_set_st *fset, socket_st *sock)
 		/* Socket buffer is full. */
 		tokens += rts->interval;
 		return MIN_INTERVAL_MS;
+	} else if (errno == EMSGSIZE) {
+		/* SOCK_DGRAM fails on -s > 65527 */
+		rts->nerrors++;
+		i = 0;
```

Comment:
Maybe I should just quit here because it does not makes sense to keep sending (it will fail anyway, kernel will refuse them all, it's not MTU based). @metan-ucw @gobenji WDYT?

I would implemented it probably like:
```c 
error(2, errno, "sendmsg");
```
which would not print statistics.

Reply:
The error message in ping4_receive_error_msg() seems to suggest that it used to be MTU based or may be MTU based in some cases, see the "error(0, 0, _("local error: message too long, mtu=%u"), e->ee_info);". That does look like a leftover though, since in my testing this error gets triggered if we cross the 64k packet size by just a small bit and it prints the network MTU which is a few order of magnitudes smaller than the maximal allowed packet size.

Reply:
I wonder myself why `CMSG_FIRSTHDR()` did not catch this. I need to search, but I guess it was for the raw socket.

Reply:
I mean it does print the correct MTU but I have no idea what is the connection to maximal ping packet size here. My guess would be that decades ago ping packet size was limited by MTU, now it's not but we still get MTU printed in the case of EMSGSIZE which does seem to be confusing.